lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3866,6 +3866,63 @@ LogicalResult ConvertAtenOp<AtenLeTensorOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenIscloseOp>::matchAndRewrite(
+    AtenIscloseOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  // check args
+  double rtol, atol;
+  bool equalNan;
+  if (!matchPattern(op.getRtol(), m_TorchConstantFloat(&rtol)))
+    return rewriter.notifyMatchFailure(op, "rtol must be a scalar constant");
+  if (!matchPattern(op.getAtol(), m_TorchConstantFloat(&atol)))
+    return rewriter.notifyMatchFailure(op, "atol must be a scalar constant");
+  if (!matchPattern(op.getEqualNan(), m_TorchConstantBool(&equalNan)))
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: equal_nan is expected to be false");
+
+  // check tensor type.
+  auto selfType = adaptor.getSelf().getType().dyn_cast<TensorType>();
+  auto otherType = adaptor.getOther().getType().dyn_cast<TensorType>();
+  if (!selfType || !otherType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
+  if (!selfType.hasStaticShape() || !otherType.hasStaticShape())
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types with static shape are supported");
+  if (!selfType.getElementType().isa<mlir::FloatType>() ||
+      !otherType.getElementType().isa<mlir::FloatType>()) {
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: only FP element type is supported");
+  }
+
+  auto rhsSubOp = rewriter.create<tosa::SubOp>(
+      op->getLoc(), selfType, adaptor.getSelf(), adaptor.getOther());
+  auto rhsAbsOp =
+      rewriter.create<tosa::AbsOp>(op->getLoc(), selfType, rhsSubOp);
+
+  auto lhsAbsOp =
+      rewriter.create<tosa::AbsOp>(op->getLoc(), otherType, adaptor.getOther());
+  auto constType =
+      RankedTensorType::get(otherType.getShape(), rewriter.getF32Type());
+  auto rtolAttr = DenseElementsAttr::get(constType, static_cast<float>(rtol));
+  auto rtolConstOp =
+      rewriter.create<tosa::ConstOp>(op->getLoc(), constType, rtolAttr);
```

Comment:
Are there any existing libraries we can use to create `tosa::ConstOp`?

Reply:
Added the e2d test.

**tosa::getTosaConstTensorSingleF32** doesn't create the the proper shape, and **tosa::getConstTensor** requires us to provide a const vector with the same size of the tensor. I think it's just easier to create a const op in place. Let me what you think. I can use the **tosa::getConstTensor** util if you think it's better.

Reply:
Why do we need to create a const of the shape corresponding to the other tensor. We should be able to create a rank-0 tensor with that const (using `getTosaConstTensorSingleF32`) and use it any binary op, since Tosa ops should support implicit broadcasting. Am I missing something here?

Reply:
Ah.. forget TOSA can do implicit broadcast. Just updated.